synthesizer/src/process/stack/mod.rs

```diff
@@ -113,8 +123,9 @@ impl<N: Network> CallStack<N> {
         authorization: Authorization<N>,
         execution: Arc<RwLock<Execution<N>>>,
         inclusion: Arc<RwLock<Inclusion<N>>>,
+        metrics: Arc<RwLock<Vec<CallMetrics<N>>>>,
```

Comment:
do we expect multiple simultaneous readers for `metrics`? if not, it would be a bit faster to use a `Mutex` here